msteams-platform/bots/how-to/format-ai-bot-messages.md

```diff
@@ -0,0 +1,433 @@
+      usageInfo: {
+        "@type": "CreativeWork",
+        "@id": "sensitivity1"
+      },
+      citation: [
+        {
+          "@type": "Claim",
+          position: 1,
+          appearance: {
+            "@type": "DigitalDocument",
+            name: "AI bot",
+            url: "https://example.com/claim-1",
+            usageInfo: {
+              "@type": "CreativeWork",
+              "@id": "sensitivity1",
+              name: "Sensitivity title",
+              description: "Sensitivity description",
+            },
+          },
+        },
+      ],
+    },
+  ],
+});
+```
+
+---
+
+| Property | Type | Required | Description |
+|--|--|--|--|
+| `usageInfo.@type` | String | Yes | Enables the sensitivity label in the bot message. |
+| `usageInfo.@id` | String | Yes | Enables referencing the sensitivity label in the citation. |
+| `usageInfo.name` | String | Yes | Specifies the title of the sensitivity label. |
+| `usageInfo.description` | String | No | Specifies the pop-up window message that appears when a user hovers over the sensitivity label. |
+
+After the sensitivity label is added, your bot message contains a shield icon. Users can hover over the icon to find a disclaimer on the sensitivity of the message.
+
+### Error handling
+
+| Error code | Description |
+| --- | --- |
+| 400 | Multiple root message entities found under `entities` array |
+| 400 | Error parsing message entity from `entities` array |
+| 400 | Citation level `usageInfo.@id` value doesn't match the message level `usageInfo.@id` in at least one instance |
+| 400 | There are multiple citation-level `usageInfo` properties with the same `@id`, but their `name` and `description` properties are different. |
+
+## Modify `PredictedSayCommand`
+
+For a bot built using Teams AI library, the `PredictedSayCommand` provides control on how AI label, sensitivity label, citations, and feedback buttons are added to the bot's activity. Following is the code snippet to modify `PredictedSayCommand`:
+
+```JavaScript
+app.ai.action<PredictedSayCommand>(AI.SayCommandActionName, async (context, state, data, action) => {
+  // custom logic here...
+  await context.sendActivity(data.content);
+  return "";
+});
+```
+
+For more information about `PredictedSayCommand`, see [PredictedSayCommand interface](/javascript/api/%40microsoft/teams-ai/predictedsaycommand?view=msteams-client-js-latest&preserve-view=true).
+
+## Code sample
+
+| Sample name | Description | Node.js |
+|:--|--|--|
+| Teams conversation bot | This sample app provides a bot with an AI label, feedback buttons, sensitivity label, and citations in messages. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/bot-conversation/nodejs) |
+
+## See also
+
+* [Bot activity handlers](../bot-basics.md)
+* [Format your bot messages](format-your-bot-messages.md)
+* [Get started with Teams AI library](Teams%20conversational%20AI/how-conversation-ai-get-started.md)
```

Comment:
These links don't appear to work for me

Reply:
Tested on multiple devices. The links are working.